source/Calamari.Terraform/ApplyCommand.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Calamari.Commands.Support;
+
+namespace Calamari.Terraform
```

Comment:
Rather than having a seperate command for every terraform step, might we be able to simply have a single step `terraform` that looks for the operation type from the variables (supplied by the ActionHandlers with something like `.WithVariable("terraform-type", "apply")`. All these commands seem to differ _only_ due to the convention being used. 

We can reduce the number of classes here if the base `TerraformCommand` simply checks what `terraform-type` value is used and supplies the appropriate convention to its chain. Ideally id even consider the type sent down by server to be a FQN that the command can just rehydrate so it doesn't need to even deal with switch statements. Remember we can currently always assume there is a direct relationship between an action handler and a specific version of Calamari.

Reply:
I see where we going but given we have plans to refactor the commands, I am somehow inclined to leave this the way it is and do the big refactor later on once we decide which way to go, does that sound reasonable for now?

Reply:
suppose i wouldn't think of it as a blocker to change it to use 1 Terraform command, although i think it simplifies things by removing a bunch of unnecessary classes that only exist to define the convention to use. Hopefully for future work we can take this into consideration. Whilst we dont need to solve the tech debt now its worthwhile to not get further from the goal.